lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -949,26 +953,34 @@ public function AliasIdentificationVariable()
     }
 
     /**
-     * AbstractSchemaName ::= identifier
+     * AbstractSchemaName ::= qualified_name | aliased_name | identifier
+     *
+     * @param bool $validateName Whether to validate that the parsed name refers to an existing class.
      *
      * @return string
      */
-    public function AbstractSchemaName()
+    public function AbstractSchemaName($validateName = true)
```

Comment:
Parser::NewObjectExpression does not want validation, at least not immediately when reading the AbstractSchemaName.

I could also extract the validation into a separate method but then you'd need to call them in several places and – most importantly – must not forget to call it when you use AbstractSchemaName. Having the validation toggleable via a parameter is ugly but errs on the side of applying too much validation.

You choose.